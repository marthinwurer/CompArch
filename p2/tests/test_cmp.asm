;
; tests CMP
;
; uses instructions MOV, BPT, HALT
;
; uses addressing modes 0, 2, 6
;
	.globl	start
;
v1:	.word	0
v2:	.word	177777
v3:	.word	077777
;
start:	mov	v1,r1
	mov	v2,r2
	mov	v3,r3
	bpt
;
	cmp	#0,r1
	cmp	#0,r2
	cmp	#0,r3
;
	cmp	r1,#0
	cmp	r2,#0
	cmp	r3,#0
;
	cmp	#0,v1
	cmp	#0,v2
	cmp	#0,v3
;
	cmp	v1,#0
	cmp	v2,#0
	cmp	v3,#0
;
	cmp	r1,r1
	cmp	r1,r2
	cmp	r1,r3
;
	cmp	r2,r1
	cmp	r2,r2
	cmp	r2,r3
;
	cmp	r3,r1
	cmp	r3,r2
	cmp	r3,r3
;
	cmp	v1,v1
	cmp	v1,v2
	cmp	v1,v3
;
	cmp	v2,v1
	cmp	v2,v2
	cmp	v2,v3
;
	cmp	v3,v1
	cmp	v3,v2
	cmp	v3,v3
;
	halt
;
	.end	start
