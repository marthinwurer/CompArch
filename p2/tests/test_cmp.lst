      1                                 ;
      2                                 ; tests CMP
      3                                 ;
      4                                 ; uses instructions MOV, BPT, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 2, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000000                  v1:      .word   0
     11 000002  177777                  v2:      .word   177777
     12 000004  077777                  v3:      .word   077777
     13                                 ;
     14 000006  016701  177766          start:   mov     v1,r1
     15 000012  016702  177764                   mov     v2,r2
     16 000016  016703  177762                   mov     v3,r3
     17 000022  000003                           bpt
     18                                 ;
     19 000024  022701  000000                   cmp     #0,r1
     20 000030  022702  000000                   cmp     #0,r2
     21 000034  022703  000000                   cmp     #0,r3
     22                                 ;
     23 000040  020127  000000                   cmp     r1,#0
     24 000044  020227  000000                   cmp     r2,#0
     25 000050  020327  000000                   cmp     r3,#0
     26                                 ;
     27 000054  022767  000000  177716           cmp     #0,v1
     28 000062  022767  000000  177712           cmp     #0,v2
     29 000070  022767  000000  177706           cmp     #0,v3
     30                                 ;
     31 000076  026727  177676  000000           cmp     v1,#0
     32 000104  026727  177672  000000           cmp     v2,#0
     33 000112  026727  177666  000000           cmp     v3,#0
     34                                 ;
     35 000120  020101                           cmp     r1,r1
     36 000122  020102                           cmp     r1,r2
     37 000124  020103                           cmp     r1,r3
     38                                 ;
     39 000126  020201                           cmp     r2,r1
     40 000130  020202                           cmp     r2,r2
     41 000132  020203                           cmp     r2,r3
     42                                 ;
     43 000134  020301                           cmp     r3,r1
     44 000136  020302                           cmp     r3,r2
     45 000140  020303                           cmp     r3,r3
     46                                 ;
     47 000142  026767  177632  177630           cmp     v1,v1
     48 000150  026767  177624  177624           cmp     v1,v2
     49 000156  026767  177616  177620           cmp     v1,v3
     50                                 ;
     51 000164  026767  177612  177606           cmp     v2,v1
     52 000172  026767  177604  177602           cmp     v2,v2
     53 000200  026767  177576  177576           cmp     v2,v3
     54                                 ;
     55 000206  026767  177572  177564           cmp     v3,v1
     56 000214  026767  177564  177560           cmp     v3,v2
     57 000222  026767  177556  177554           cmp     v3,v3
     58                                 ;
     59 000230  000000                           halt
     60                                 ;
     61 000232  000006'                          .end    start


pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000006 RG 
v1       000000 R  
v2       000002 R  
v3       000004 R  
