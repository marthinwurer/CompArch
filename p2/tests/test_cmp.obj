000 2 00 00
002 2 ff ff
004 2 ff 7f
006 4 c1 1d f6 ff
00a 4 c2 1d f4 ff
00e 4 c3 1d f2 ff
012 2 03 00
014 4 c1 25 00 00
018 4 c2 25 00 00
01c 4 c3 25 00 00
020 4 57 20 00 00
024 4 97 20 00 00
028 4 d7 20 00 00
02c 6 f7 25 00 00 ce ff
032 6 f7 25 00 00 ca ff
038 6 f7 25 00 00 c6 ff
03e 6 d7 2d be ff 00 00
044 6 d7 2d ba ff 00 00
04a 6 d7 2d b6 ff 00 00
050 2 41 20
052 2 42 20
054 2 43 20
056 2 81 20
058 2 82 20
05a 2 83 20
05c 2 c1 20
05e 2 c2 20
060 2 c3 20
062 6 f7 2d 9a ff 98 ff
068 6 f7 2d 94 ff 94 ff
06e 6 f7 2d 8e ff 90 ff
074 6 f7 2d 8a ff 86 ff
07a 6 f7 2d 84 ff 82 ff
080 6 f7 2d 7e ff 7e ff
086 6 f7 2d 7a ff 74 ff
08c 6 f7 2d 74 ff 70 ff
092 6 f7 2d 6e ff 6c ff
098 2 00 00
006
