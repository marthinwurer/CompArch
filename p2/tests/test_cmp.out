CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 6
000006:  PS[00]  016701 177766  MOV 177766(R7),R1
000012:  PS[04]  016702 177764  MOV 177764(R7),R2
000016:  PS[10]  016703 177762  MOV 177762(R7),R3
000022:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[077777]
  R4[000000]  R5[000000]  R6[000000]  R7[000024]
000024:  PS[00]  022701  CMP (R7)+,R1
000030:  PS[04]  022702  CMP (R7)+,R2
000034:  PS[01]  022703  CMP (R7)+,R3
000040:  PS[11]  020127  CMP R1,(R7)+
000044:  PS[04]  020227  CMP R2,(R7)+
000050:  PS[10]  020327  CMP R3,(R7)+
000054:  PS[00]  022767 177716  CMP (R7)+,177716(R7)
000062:  PS[04]  022767 177712  CMP (R7)+,177712(R7)
000070:  PS[01]  022767 177706  CMP (R7)+,177706(R7)
000076:  PS[11]  026727 177676  CMP 177676(R7),(R7)+
000104:  PS[04]  026727 177672  CMP 177672(R7),(R7)+
000112:  PS[10]  026727 177666  CMP 177666(R7),(R7)+
000120:  PS[00]  020101  CMP R1,R1
000122:  PS[04]  020102  CMP R1,R2
000124:  PS[01]  020103  CMP R1,R3
000126:  PS[11]  020201  CMP R2,R1
000130:  PS[10]  020202  CMP R2,R2
000132:  PS[04]  020203  CMP R2,R3
000134:  PS[10]  020301  CMP R3,R1
000136:  PS[00]  020302  CMP R3,R2
000140:  PS[13]  020303  CMP R3,R3
000142:  PS[04]  026767 177632 177630  CMP 177632(R7),177630(R7)
000150:  PS[04]  026767 177624 177624  CMP 177624(R7),177624(R7)
000156:  PS[01]  026767 177616 177620  CMP 177616(R7),177620(R7)
000164:  PS[11]  026767 177612 177606  CMP 177612(R7),177606(R7)
000172:  PS[10]  026767 177604 177602  CMP 177604(R7),177602(R7)
000200:  PS[04]  026767 177576 177576  CMP 177576(R7),177576(R7)
000206:  PS[10]  026767 177572 177564  CMP 177572(R7),177564(R7)
000214:  PS[00]  026767 177564 177560  CMP 177564(R7),177560(R7)
000222:  PS[13]  026767 177556 177554  CMP 177556(R7),177554(R7)
000230:  PS[04]  000000  HALT

Machine Halted - HALT instruction

Simulated time 369 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
